File: backend/BookstoreTamAn/src/main/java/com/minhnghia/datn/BookstoreTamAn/model/Book.java
package com.minhnghia.datn.BookstoreTamAn.model;


import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.Date;
import java.util.Set;

@Entity
@Table(name = "book")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Book extends BaseEntity{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String title;
    private String format;
    private Integer pages;
    private Float width;
    private Float height;
    private Float length;
    private Float weight;
    private Date publicationDate;
    private String publisher;
    private String language;
    private String illustrationsNote;
    private String ISBN10;
    private String ISBN13;
    private Double promotion;
    private LocalDate promotionEndDate;
    private String description;
    private String sellerReview;
    private String reviewVideo;
    private Double price;
    private Integer stock;
    private String imageUrl;
    private Float rating;
    private Integer numReviews;
    private Integer selling;
    private String createdBy;
    private String modifyBy;

    @ManyToOne
    @JoinColumn(name = "author_id", nullable = false)
    private Author author;

    @ManyToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinTable(
            name = "book_category",
            joinColumns = @JoinColumn(name = "book_id"),
            inverseJoinColumns = @JoinColumn(name = "category_id")
    )
    private Set<Category> categories;
}

File: frontend/frontend/src/app/models/book.ts
import { Category } from "./category";
import { CategoryNames } from "./categoryNames";

export class Book{
    id : 0;
    title: string;
    format : string;
    pages : 0;
    width : 0;
    height : 0;
    length : 0;
    weight : 0;
    publicationDate : string;
    publisher : string;
    language : string;
    illustrationsNote: string;
    isbn10: string;
    isbn13 : string;
    promotion : 0;
    promotionEndDate : string;
    description: string;
    sellerReview : string;
    reviewVideo : string;
    price : 0;
    stock: 0;
    imageUrl: string;
    rating: 0;
    numReviews: 0;
    selling: 0;
    createdBy: string;
    modifyBy: string;
    createdAt: string;
    modifyAt: string;
    nameAuthor : string;
    authorId : 0;
    categoryNames: Category[] = [];
    active : boolean =  true;

    constructor(){
        this.id = 0;
        this.title = "";
        this.format = "";
        this.pages = 0;
        this.width = 0;
        this.height = 0;
        this.length = 0;
        this.weight = 0;
        this.publicationDate = "";
        this.publisher = "";
        this.language = "";
        this.illustrationsNote = "";
        this.isbn10 = "";
        this.isbn13 = "";
        this.promotion = 0,
        this.promotionEndDate = "";
        this.description = "";
        this.sellerReview = "";
        this.reviewVideo = "",
        this.price = 0;
        this.stock = 0;
        this.imageUrl = "";
        this.rating = 0;
        this.numReviews = 0;
        this.selling = 0;
        this.createdBy = "";
        this.modifyBy = "";
        this.createdAt = "";
        this.modifyAt = "";
        this.nameAuthor = "";
        this.authorId = 0;
    }
}

export interface IBookList{
    id : 0;
    title: string;
    format : string;
    pages : 0;
    width : 0;
    height : 0;
    length : 0;
    weight : 0;
    publicationDate : string;
    publisher : string;
    language : string;
    illustrationsNote: string;
    isbn10: string;
    isbn13 : string;
    promotion : 0;
    promotionEndDate : string;
    description: string;
    sellerReview : string;
    reviewVideo : string;
    price : 0;
    stock: 0;
    imageUrl: string;
    rating: 0;
    numReviews: 0;
    selling: 0;
    createdBy: string;
    modifyBy: string;
    createdAt: string;
    modifyAt: string;
    nameAuthor : string;
    authorId : 0;
    categoryNames : Category[];
    active : boolean;
}